Update Docker deployment scripts

danlamanna opened the pull request:
This is just a new Dockerfile, soon to come is the infrastructure for deploying this with SMQTK.

danlamanna committed:
Update Docker deployment scripts

danlamanna commented:
To test:
```
docker network create imagespace-network
docker run --name imagespace-mongo -d --net=imagespace-network mongo:3.0
docker run --name imagespace-girder -P --net=imagespace-network kitware/image_space --database mongodb://imagespace-mongo/girder
```

Then visit localhost at whatever port `imagespace-girder` got mapped to (login is girder/girder).

Note you won't be able to really do anything since the Solr component is forthcoming, but it should give you access to a Girder installation with a user and ImageSpace installed/enabled.

danlamanna committed:
Add solr import script

danlamanna committed:
Add mounted image volume to girder container

chrismattmann commented on scripts/deploy/docker-compose.yml:
```diff
@@ -0,0 +1,26 @@
+      - solr-precreate
+      - imagespace
+  imagespace-girder:
+    image: kitware/image_space:latest
```
why does this depend on something specific to kitware? Shouldn't it be memex-explorer/image_space:latest?

chrismattmann commented on scripts/deploy/imagespace/Dockerfile:
```diff
@@ -0,0 +1,29 @@
+# Docker image: kitware/imagespace
```
should be memex-explorer/imagespace

chrismattmann commented on scripts/deploy/imagespace/Dockerfile:
```diff
@@ -0,0 +1,29 @@
+
+# Requirements:
+# - Directory of images to mount for serving (at IMAGE_SPACE_IMAGE_DIR)
+# - Solr instance with URL to core as IMAGE_SPACE_SOLR
```
please add ALv2 license header.

chrismattmann commented on scripts/deploy/docker-compose.yml:
```diff
@@ -0,0 +1,26 @@
+version: "2"
```
please add ALv2 license header.

chrismattmann commented on scripts/deploy/imagespace/imagespace-entrypoint.sh:
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
please add ALv2 license header.

chrismattmann commented on scripts/deploy/imagespace/bootstrap-imagespace.py:
```diff
@@ -0,0 +1,44 @@
+import json
```
please add ALv2 license header.

chrismattmann commented:
@danlamanna some simple cleanup and I think this is ready to go I'll try and test this week.

jeffbaumes committed:
Exposing the imagespace app port

jeffbaumes commented:
@danlamanna I was able to access it when I exposed the port to the host, see my latest commit (i randomly chose port 8989 since 8080 is common).

chrismattmann commented:
+1 @jeffbaumes 👍 on the port

danlamanna committed:
Adding license headers

danlamanna commented:
Fixed up some of these, but I'm setting this to WIP for the moment - needs some more work.

danlamanna committed:
Update Solr import script to use post utility

danlamanna committed:
Update Girder container to wait for Mongo

danlamanna committed:
Use memexexplorer docker image

danlamanna committed:
Add readme docs

jeffbaumes committed:
Fixes for mac

--max-procs is not supported on Mac xargs, and -P 0 does not work either (requires > 0)

Instead, using cross-platform way to detect cores http://stackoverflow.com/questions/1715580/how-to-discover-number-of-logical-cores-on-mac-os-x

Also sed -i is not working, added ''. See http://stackoverflow.com/questions/7573368/in-place-edits-with-sed-on-os-x

jeffbaumes commented:
@danlamanna check that the fixes for Mac don't break Linux. Mac's Unix tools are messed up.

jeffbaumes committed:
Add Mac install note

danlamanna committed:
Remove usage of sed and extra temporary file

danlamanna commented:
@jeffbaumes It did break Linux, check if my new changes work on Mac?

jeffbaumes commented:
Works! LGTM
